### PR TITLE
[Bug](type) fix wildcard char's tostring get wrong result

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

support parse char(*)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -6311,6 +6311,8 @@ type ::=
   {: RESULT = ScalarType.createVarcharType(-1); :}
   | KW_VARCHAR
   {: RESULT = ScalarType.createVarcharType(-1); :}
+  | KW_CHAR LPAREN STAR RPAREN
+  {: RESULT = ScalarType.createCharType(-1); :}
```
Comment (morrySnow) [inline fe/fe-core/src/main/cup/sql_parser.cup:6315]: maybe we should let char type to sql return `char` only if length = -1

**Comment by morrySnow** [comment]
btw, why we generate char(-1) type? char should always with fix length

**Comment by BiteTheDDDDt** [comment]
we will do not support char(*)

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**